src/polonius.dl

```diff
@@ -62,31 +62,52 @@
 //          * If `P = *Q` where `Q: &'Q mut T`, then `'Q: 'O` and continue "unrolling" `Q`
 // * Generate `introduce_subset` for subsets that arise from other where-clauses or type relations on this node
 
-// access_origin data with origin `o` at node `n`
+// Indicates that a reference with origin `o` was dereferenced (or may be).
+//
+// Issued for all origins in the type of some place `P` when `P` is read.
 .decl access_origin(o: Origin, n: Node)
 .input access_origin
 
-// Invalidate the lease `l` at the node `n`
+// Invalidates data potentially derived from the lease `l` at the node `n`.
+//
+// Issued when the place that `l` is leased from is accessed in some incompatible
+// way (e.g., if `l` is an `&mut` lease of some place `P`, then `l` is invalidated
+// when `P` is accessed; if `l` is a `&` lease of `P`, then it is invalidated
+// when `P` is written).
 .decl invalidate_lease(l: Lease, n: Node)
 .input invalidate_lease
 
-// All references with origin `o` are overwritten at node `n`
+// All values with origin `o` in their type are overwritten at node `n`.
+//
+// Since the compiler uses unique origins in the type of all values,
+// this is used when you for all origins in the type of `place` when you
+// have an assignment like `place = ...`. 
 .decl clear_origin(o: Origin, n: Node)
 .input clear_origin
 
-// Lease `l` removed from all origins
+// The path that lease `l` referred to now refers to different memory.
+//
+// This is issued for a lease to `*x` when `x` is overwritten.
 .decl clear_lease(l: Lease, n: Node)
 .input clear_lease
 
-// Require that `o1 <= o2` at node `n`
+// Require that `o1: o2` at node `n`.
+//
+// This fact indicates that a given node requires a subset relation to hold. 
+// Most notably, `&'a T <: &'b T` introduces a requirement that `'a: 'b`
+// (`'a <= 'n`).
```

Comment:
what's `'n` here ? 

do you mean that `'a` contains the node `n` in the sets-of-points formulation ?

Reply:
lol no. it's a typo.

Reply:
```suggestion
// (`'a <= 'b`).
```

Reply:
if `'n` is `'b`, it's the shorter one then, shouldn't that be `'b <= 'a`